pkg/transparentproxy/kubernetes/kubernetes.go

```diff
@@ -115,25 +128,27 @@ func NewPodRedirectForPod(pod *kube_core.Pod) (*PodRedirect, error) {
 
 func (pr *PodRedirect) AsTransparentProxyConfig() *config.TransparentProxyConfig {
 	return &config.TransparentProxyConfig{
-		DryRun:                 false,
-		Verbose:                true,
-		RedirectPortOutBound:   fmt.Sprintf("%d", pr.RedirectPortOutbound),
-		RedirectInBound:        pr.RedirectInbound,
-		RedirectPortInBound:    fmt.Sprintf("%d", pr.RedirectPortInbound),
-		RedirectPortInBoundV6:  fmt.Sprintf("%d", pr.RedirectPortInboundV6),
-		ExcludeInboundPorts:    pr.ExcludeInboundPorts,
-		ExcludeOutboundPorts:   pr.ExcludeOutboundPorts,
-		UID:                    pr.UID,
-		GID:                    pr.UID, // TODO: shall we have a separate annotation here?
-		RedirectDNS:            pr.BuiltinDNSEnabled,
-		RedirectAllDNSTraffic:  false,
-		AgentDNSListenerPort:   fmt.Sprintf("%d", pr.BuiltinDNSPort),
-		DNSUpstreamTargetChain: "",
-		ExperimentalEngine:     pr.ExperimentalTransparentProxyEngine,
-		EbpfEnabled:            pr.TransparentProxyEnableEbpf,
-		EbpfInstanceIP:         pr.TransparentProxyEbpfInstanceIPEnvVarName,
-		EbpfBPFFSPath:          pr.TransparentProxyEbpfBPFFSPath,
-		EbpfProgramsSourcePath: pr.TransparentProxyEbpfProgramsSourcePath,
+		DryRun:                         false,
+		Verbose:                        true,
+		RedirectPortOutBound:           fmt.Sprintf("%d", pr.RedirectPortOutbound),
+		RedirectInBound:                pr.RedirectInbound,
+		RedirectPortInBound:            fmt.Sprintf("%d", pr.RedirectPortInbound),
+		RedirectPortInBoundV6:          fmt.Sprintf("%d", pr.RedirectPortInboundV6),
+		ExcludeInboundPorts:            pr.ExcludeInboundPorts,
+		ExcludeOutboundPorts:           pr.ExcludeOutboundPorts,
+		ExcludeOutboundTCPPortsForUIDs: pr.ExcludeOutboundTCPPortsForUIDs,
+		ExcludeOutboundUDPPortsForUIDs: pr.ExcludeOutboundUDPPortsForUIDs,
+		UID:                            pr.UID,
+		GID:                            pr.UID, // TODO: shall we have a separate annotation here?
+		RedirectDNS:                    pr.BuiltinDNSEnabled,
+		RedirectAllDNSTraffic:          false,
+		AgentDNSListenerPort:           fmt.Sprintf("%d", pr.BuiltinDNSPort),
+		DNSUpstreamTargetChain:         "",
+		ExperimentalEngine:             pr.ExperimentalTransparentProxyEngine,
+		EbpfEnabled:                    pr.TransparentProxyEnableEbpf,
+		EbpfInstanceIP:                 pr.TransparentProxyEbpfInstanceIPEnvVarName,
+		EbpfBPFFSPath:                  pr.TransparentProxyEbpfBPFFSPath,
+		EbpfProgramsSourcePath:         pr.TransparentProxyEbpfProgramsSourcePath,
 	}
 }
```

Comment:
> I think you need to add these new entries to AsKumactlCommandLine() instead.

While you were writing this I realised that it actually did not work for k8s and started [implementing that](https://github.com/kumahq/kuma/pull/4975/commits/4d4b3c08ffaeb2d0b92d239aa342f6171a43b176) :)

>  Also would it make sense to have AsKumactlCommandLine() be a method on config.TransparentProxy maybe?

I think that `PodRedirect` is just redundant. Couldn't we just get `config.TransparentProxy` from Annotations instead of having a middle man? I'll do this but I'd like to do it in a separate PR just to make sure I'm not breaking anything.

Reply:
Created an issue for this: https://github.com/kumahq/kuma/issues/4986